### PR TITLE
chore(KFLUXVNGD-142): support contexts in its

### PR DESCRIPTION
**Pull request by yftacherzog**
Add support for parameterizing name and description fields inside the IntegrationTestScenario contexts list

### CODE REVIEW

File: internal/template/resources.go [hunk 1]
```diff
@@ -104,6 +104,8 @@ var supportedResourceTypes = []struct {
 		},
 		templateAbleFields: [][]string{
 			{"spec", "resolverRef", "params", "[]", "value"},
+			{"spec", "contexts", "[]", "name"},
+			{"spec", "contexts", "[]", "description"},
```
Comment (ifireball) [inline internal/template/resources.go:108]: Should this be a name field?

File: config/samples/projctl_v1beta1_pdst_w_intgtstscnario.yaml [hunk 2]
```diff
@@ -57,6 +57,8 @@ spec:
       contexts:
         - description: Application testing
           name: application
+        - description: Application testing for cool-app-{{.versionName}}
+          name: component_cool-app-{{.versionName}}
```
Comment (ifireball) [inline config/samples/projctl_v1beta1_pdst_w_intgtstscnario.yaml:61]: is this a resource name?

File: internal/template/resources.go (hunk 1 above)
Comment (yftacherzog) [reply internal/template/resources.go:108]: It's not a name field. It's a string with no other restrictions.
Also, the specific pattern that we're trying to support under this original request is `component_COMPONENT`. I.e. it should contain underscore after the literal `component` and before the name of the component, which does not comply with a name field.

File: config/samples/projctl_v1beta1_pdst_w_intgtstscnario.yaml (hunk 2 above)
Comment (yftacherzog) [reply config/samples/projctl_v1beta1_pdst_w_intgtstscnario.yaml:61]: It's not

**Review by ifireball — APPROVED** [review]

**Pull request merged**